Wikipedia/Code/ViewControllerRouter.swift

```diff
@@ -26,6 +26,10 @@ class ViewControllerRouter: NSObject {
             }
         }
 
+        if let vc = viewController as? SinglePageWebViewController, navigationController.modalPresentationStyle == .pageSheet {
+            vc.navigationBar.isBarHidingEnabled = false
+        }
+
```

Comment:
I get slightly worried when high level specific logic seeps into these lower level controllers, especially in the case of something possibly temporary (like this experiment). What do you think of something like this below to be a bit more defensive regarding the cases the custom logic is applied to?

```suggestion
if #available(iOS 13.0, *), navigationController.children.contains(where: { $0 is ArticleAsLivingDocViewController }) {
	// All article as living document specific behavior
	if UIDevice.current.userInterfaceIdiom == .pad, let vc = viewController as? SinglePageWebViewController, navigationController.modalPresentationStyle == .pageSheet {
		vc.navigationBar.isBarHidingEnabled = false
	}
}
```

Reply:
This is great, thank you. I really cringed at writing this, but the way the router is written I couldn't do it earlier. Adding more checks to this is good, agreed. Thanks.